services/src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -181,7 +183,11 @@ public void start() {
   @Override
   public void stop() {
     if (_running.compareAndSet(true, false)) {
-      _consumer.close();
+      try {
+        _consumer.close();
+      } catch (ConcurrentModificationException cme) {
```

Comment:
Would it be better to cache all exceptions instead of simply the ConcurrentModificationException? The idea is that we want to log all exceptions that may be thrown out of `close()`.